RFCs/0025-graph-data-model.md

```diff
@@ -54,51 +63,99 @@ Which is similarly extracted as variables:
-and edge are a property bag which is a struct in PartiQL.
+As an important example, *labeled property graphs (LPG)* of the GPML paper[^2] are modeled 
+straightforwardly in PartiQL: a value at each vertex and edge is restricted to be a PartiQL struct.
 
+<!-- TODO: The following RDF example may need adjustments to fit with the finalized data model.
+     E.g., NULL labels may have to become empty label sets.  -->
 Similarly, Resource Description Framework (RDF)[^1] graphs could be modeled in PartiQL by having non-literal, non-blank
 vertices and edges labeled by URI strings with their values being NULL. RDF literals could be a NULL labeled node with
 their value being any corresponding PartiQL value (this is a generalization of RDF as literals are only strings in RDF’s
 model). RDF blank nodes can be denoted with a label that never conflicts with URI such as **`_:my_blank`** (**`_`** is
 never a valid scheme for a URI).
 
-Even though PartiQL generalizes the graph data model, it is *not required* that a database actually exposes arbitrary
-values at vertex or edge values, this is similar to PartiQL over a relational table, where attributes of a row are
+Even though PartiQL defines a very general graph data model, it is *not required* that a host database actually supports arbitrary
+values at vertices or edges. This is similar to PartiQL over a relational database, where attributes of a row are
 restricted to scalars.
 
-## Data Model Integration with PartiQL Query
+## Addition to the PartiQL data model
 
-The SQL/PGQ specification is currently in progress, but has published elements of their work[^2]. It is important that
-PartiQL aligns to an SQL standard that arises around graph query insofar as it is practically acceptable. Let us
-consider a straw example of what a PartiQL graph query could look like and mean with respect to this data model.
+As outlined above, graphs are introduced as a new category of "native" values in the PartiQL
+data model, on par with scalar literals, structs, lists, and bags:
 
-```sql
-SELECT the_a.name AS src, the_b.name AS dest
-FROM my_graph MATCH (the_a:a) -[the_y:y]-> (the_b:b)
-WHERE the_y.score > 10
+```EBNF
+<partiql value> ::= 
+      <absent value>
+    | <scalar value>
+    | <tuple value>
+    | <collection value>
+    | <graph value>                  //new
 ```
+Consequently, graph values can occur as members of structs, bags, and lists.
+
+## Graph data model
+
+For the graph values themselves, rather than giving a grammar -- as PartiQL specification does for
+other values -- we will define an abstract data model, largely following the one for PGML[^2].
+One of the reasons is that a concrete syntax for graph literals could emerge from ISO/IEC
+standardization, possibly as part of GQL, but no public information is yet available about
+that effort.  When appropriate, graph literals in PartiQL would be covered in a separate RFC.
```

Comment:
Minor edits suggested below.  It is "GPML" versus "PGML", right?

```suggestion
For graph values themselves, rather than giving a grammar--as the PartiQL specification does for
other values--we will define an abstract data model, largely following the one for GPML[^2].
We do this because a concrete syntax for graph literals could emerge from ISO/IEC
standardization, possibly as part of GQL, but no public information is yet available about
that effort.  When appropriate, graph literals in PartiQL would be covered in a separate RFC.
```